polling_stations/apps/data_importers/management/commands/import_eoni_from_s3.py

```diff
@@ -0,0 +1,415 @@
+    else:
+        return body
+    lines = body.split(line_sep)
+    if len(lines) > 20:
+        return line_sep.join(lines[:20])
+    return line_sep.join(lines)
+
+
+def is_eoni_export_key(key: str) -> bool:
+    """
+    Should return true for things like:
+    '2023-04-12-EONIextract.txt',
+    '2024-07-03-EONIextract-sample.txt',
+    '2024-11-21-EONIextract_NEW.txt'
+
+    """
+    eoni_export_key_pattern = re.compile(
+        r"""
+        ^           # Start of string
+        20[2-9]     # First 3 digits of year: 202-209
+        [0-9]       # Last digit of year: 0-9
+        -           # Separator
+        [0-9]{2}    # Month as 2 digits
+        -           # Separator
+        [0-9]{2}    # Day as 2 digits
+        -.+         # Hyphen followed by any text
+        \.txt       # Literal .txt extension
+        """,
+        re.VERBOSE,
+    )
+    if eoni_export_key_pattern.match(key):
+        return True
+    return False
+
+
+class EONIUploadValidationError(Exception):
+    pass
+
+
+class Command(BaseCommand):
+    help = (
+        "Downloads EONI data from S3 and:"
+        "\n  - reprojects coordinates from EPSG:29902 to EPSG:4326"
+        "\n  - controls the eoni election scheme that is imported"
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.dc_environment = None
+        self.output_path = None
+        self.address_non_coord_fields = [
+            "PRO_ID",
+            "PRO_LINE1",
+            "PRO_LINE2",
+            "PRO_LINE3",
+            "PRO_LINE4",
+            "PRO_LINE5",
+            "PRO_POSTCODE",
+            "PRO_FULLADDRESS",
+            "PRO_UPRN",
+        ]
+        self.address_x_field = "PRO_X_COR"
+        self.address_y_field = "PRO_Y_COR"
+        self.national_suffix = "_WM"
+        self.local_suffix = "_LC"
+        self.station_non_coord_fields = [
+            "PREM_ID",
+            "PREM_NAME",
+            "PREM_LINE1",
+            "PREM_LINE2",
+            "PREM_LINE3",
+            "PREM_LINE4",
+            "PREM_LINE5",
+            "PREM_POSTCODE",
+            "PREM_FULLADDRESS",
+            "PREM_UPRN",
+        ]
+        self.station_x_field_prefix = "PREM_X_COR"
+        self.station_y_field_prefix = "PREM_Y_COR"
+
+        self.national_station_fields = [
+            field + self.national_suffix
+            for field in self.station_non_coord_fields
+            + [f"{self.station_x_field_prefix}", f"{self.station_y_field_prefix}"]
+        ]
+        self.local_station_fields = [
+            field + self.local_suffix
+            for field in self.station_non_coord_fields
+            + [f"{self.station_x_field_prefix}", f"{self.station_y_field_prefix}"]
+        ]
+        self.header = (
+            self.address_non_coord_fields
+            + [self.address_x_field, self.address_y_field]
+            + self.national_station_fields
+            + self.local_station_fields
+        )
```

Comment:
Maybe these should be globals? dunno if it matters.

Reply:
I don't see an issue with declaring them inside the class tbh